corgi/tasks/prod_defs.py

```diff
@@ -25,6 +28,64 @@
 RE_VERSION_LIKE_STRING = re.compile(r"\d[\dz.-]*$|$")
 
 
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def slow_update_builds_for_variant(
+    variant_name: str,
+    updated_stream: tuple[str, str],
+    updated_version: Optional[tuple[str, str]] = None,
+    updated_product: Optional[tuple[str, str]] = None,
+):
+    """Ensures new stream parent of the passed in variant is reflected in all the variant's builds
+    and child components"""
+    logger.info(
+        f"Updating components related to {variant_name} with new stream details: {updated_stream}"
+    )
+    updated_stream_objs = (
+        ProductStream.objects.get(name=updated_stream[0]),
+        ProductStream.objects.get(name=updated_stream[1]),
+    )
+    if updated_version:
+        logger.info(f"Also updating product_version with {updated_version}")
+        updated_version_objs = (
+            ProductVersion.objects.get(name=updated_version[0]),
+            ProductVersion.objects.get(name=updated_version[1]),
+        )
+    if updated_product:
+        logger.info(f"Also updating product {updated_product}")
+        updated_product_objs = (
+            Product.objects.get(name=updated_product[0]),
+            Product.objects.get(name=updated_product[1]),
+        )
+    for software_build in (
+        SoftwareBuild.objects.filter(relations__product_ref=variant_name).distinct().iterator()
+    ):
+        for component in software_build.components.get_queryset().iterator():
```

Comment:
I think this can be simplified now that only root components are linked to builds:
```suggestion
        root_component = software_build.components.get():
```

Reply:
ok, fixed.